themes/default/content/blog/announcing-kubernetes-helm-release-resource-preview/index.md

```diff
@@ -0,0 +1,305 @@
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+
+		namespace, err := corev1.NewNamespace(ctx, "redis-ns", nil)
+		if err != nil {
+			return err
+		}
+
+		redisPassword, err := random.NewRandomPassword(ctx, "pass", &random.RandomPasswordArgs{
+			Length: pulumi.Int(10),
+		})
+		if err != nil {
+			return err
+		}
+
+		rel, err := helm.NewRelease(ctx, "redis-helm", &helm.ReleaseArgs{
+			Chart: pulumi.String("redis"),
+			RepositoryOpts: helm.RepositoryOptsArgs{
+				Repo: pulumi.String("https://charts.bitnami.com/bitnami"),
+			},
+			Version:   pulumi.String("13.0.0"),
+			Namespace: namespace.Metadata.Name(),
+
+      // Values from Chart's parameters specified hierarchically,
+      // see https://artifacthub.io/packages/helm/bitnami/redis/13.0.0#parameters for reference.
+			Values: pulumi.Map{
+				"cluster": pulumi.Map{
+					"enabled": pulumi.Bool(true),
+				},
+				"metrics": pulumi.Map{
+					"enabled": pulumi.Bool(true),
+					"service": pulumi.Map{
+						"annotations": pulumi.StringMap{
+							"prometheus.io/port": pulumi.String("9127"),
+						},
+						"type": pulumi.String("ClusterIP"),
+					},
+				},
+				"global": pulumi.Map{
+					"redis": pulumi.Map{
+						"password": redisPassword,
+					},
+				},
+				"rbac": pulumi.BoolMap{
+					"create": pulumi.Bool(true),
+				},
+			},
+
+			// By default Release resource will wait till all created resources
+			// are available. Set this to true to skip waiting on resources being
+			// available.
+			SkipAwait: pulumi.BoolPtr(false),
+		})
+		if err != nil {
+			return err
+		}
+
+		// We can lookup resources once the release is installed. The releases'
+		// status field is set once the installation completes, so this, combined
+		// with `skipAwait: pulumi.BoolPtr(false)` above, will wait to retrieve
+    //  the Redis master ClusterIP till all resources in the Chart are available.
+		svc := pulumi.All(rel.Status.Namespace(), rel.Status.Name()).
+			ApplyT(func(r interface{}) (interface{}, error) {
+				arr := r.([]interface{})
+				namespace := arr[0].(*string)
+				name := arr[1].(*string)
+				svc, err := corev1.GetService(ctx,
+					"redis-master-svc",
+					pulumi.ID(fmt.Sprintf("%s/%s-redis-master", *namespace, *name)),
+					nil,
+				)
+				if err != nil {
+					return "", nil
+				}
+				return svc.Spec.ClusterIP(), nil
+			})
+		ctx.Export("redisMasterClusterIP", svc)
+
+		return nil
+	})
+}
+
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+import pulumi
+from pulumi import Output
+from pulumi_random.random_password import RandomPassword
+from pulumi_kubernetes.core.v1 import Namespace, Service
+from pulumi_kubernetes.helm.v3 import Release, ReleaseArgs, RepositoryOptsArgs
+
+namespace = Namespace("redis-ns")
+
+redis_password = RandomPassword("pass", length=10)
+release = Release("redis-helm",
+                  args=ReleaseArgs(
```

Comment:
Can we format this code so that there's less space at the left and more room for the useful code?

Reply:
Updated.